chore: add .env.example for easier environment setup and refine .gitignore

piotrski opened the pull request:
I’ve made a couple of updates here to developer experience:
 - Added `.env.example` for easier environment setup
This provides a template for environment variables, helping developers get their local setup configured quickly.

- Updated `.gitignore` to exclude Azurite local storage files
We’ve added patterns to ignore Azurite's local storage files `(__azurite_db_*, __blobstorage__, __queuestorage__)`, keeping the working directory cleaner during local development.

piotrski commented:
@microsoft-github-policy-service agree company="Callstack"

lucen-ms commented on api/.env.example:
```diff
@@ -0,0 +1,70 @@
+# To enable generating diffs for releases
+ENABLE_PACKAGE_DIFFING=false
+
+# To enable KeyVault credential resolution
```
Something is missing here.

lucen-ms commented on api/README.md:
`````diff
@@ -16,11 +16,26 @@ Additionally, you need to specify [EMULATED](ENVIRONMENT.md#emulated) flag equal
+   ````bash
+   cp .env.example .env
+   ````
+   Fill in the values for each environment variable in the `.env` file according to your development or production setup.
`````
Only mandatory variables are needed - please mark mandatory values in `.env.example` file.

sanduluca replied:
It’s best to list all environment variables that the app uses, even if some are optional. This ensures transparency and clarity for developers who may work on or deploy the application in different environments.

How exactly should be "marked" the mandatory values ?

lucen-ms replied:
Up to you - e.g.

```
# Required, the URL of your server
SERVER_URL=http://localhost:3000
```

lucen-ms commented:
@piotrski will you be finishing the PR please?

piotrski commented:
@lucen-ms apologies, it must have slipped my mind. I'll take a look at this later today.

piotrski commented:
I've updated the formatting of the .env.example file to make it cleaner and more organized. Additionally, I've added information indicating which variables are required, as you requested.

Let me know if you have any feedback or if further adjustments are needed!

lucen-ms commented on api/.env.example:
```diff
@@ -0,0 +1,74 @@
+# ==============================
+TEST_AZURE_STORAGE=false  # Run API unit tests against Azure storage
+AZURE_STORAGE_ACCOUNT=    # Storage account for testing (if TEST_AZURE_STORAGE=true)
+AZURE_STORAGE_ACCESS_KEY= # Storage access key for testing
```
AZURE_STORAGE_ACCOUNT=    # Storage account for testing (if TEST_AZURE_STORAGE=true)
    AZURE_STORAGE_ACCESS_KEY= # Storage access key for testing

This is duplication of the same settings above which will override them. Every setting should be only once in the example.

piotrski replied:
You're right, I just removed them.

lucen-ms commented:
Thank you @piotrski for the contribution - I've merged your changes.
